.github/workflows/registration.yml

```diff
@@ -0,0 +1,71 @@
+name: Registration
+
+on:
+  push:
+    branches:
+      - "**"
+    paths:
+      - registration/**
+      - common/**
+      - commoneventconsumer/**
+      - commontest/**
+      - cdk/**
+      - project/**
+      - build.sbt
+      - .github/workflows/registration.yml
+  workflow_dispatch:
+
+# allow queued workflows to interrupt previous runs
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.number || github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-java@v3
+        with:
+          distribution: corretto
+          cache: sbt
+          java-version: 11
+
+      - uses: actions/setup-node@v3
+        with:
+          cache: npm
+          cache-dependency-path: cdk/package-lock.json
+          node-version-file: .nvmrc
+
+      - uses: aws-actions/configure-aws-credentials@v2
+        with:
+          aws-region: eu-west-1
+          role-to-assume: ${{ secrets.GU_RIFF_RAFF_ROLE_ARN }}
+
+      - name: CDK synth
+        run: |
+          cd cdk
+          npm install
+          npm run lint
+          npm test
```

Comment:
Since you're writing tests, might be nice to add a test reporter step after this.
This is the one I've used:
https://github.com/dorny/test-reporter/tree/main

and here's an example:
https://github.com/guardian/janus/blob/e41b96a9221c0f2cf5d558990d81b3d8dd981bbc/.github/workflows/build.yml#L55-L64

Reply:
I've been looking at the test reports - for the scala tests we use specs2, which output test data in the .stats format. I don't think this is supported by the action. I tested anyway and got an expected [error](https://github.com/guardian/mobile-n10n/actions/runs/5961506739/job/16170846330).

For the cdk tests, I can't see the output of the tests on my local machine, so not entirely sure what path to specify for the reporter. Will do a little more digging.